controller/api-controllers.js

```diff
@@ -47,4 +46,22 @@ exports.getArticleCommentsById = (req, res, next) => {
         res.status(200).send({ comments });
     })
     .catch(next);
+};
+
+exports.postCommentByArticleId = (req, res, next) => {
+    if(Object.keys(req.body).length === 0){
+        res.status(400).send({msg: "Bad Request: comment empty"});
+    }
+    else if(typeof req.body.username !== "string" || typeof req.body.body !== "string"){
+        res.status(400).send({msg: "Bad Request"});
+    }
+    const { article_id } = req.params;
+    if(!Number(article_id)){
+        res.status(400).send({msg:"Bad Request: article_id must be a number"});
+    }
```

Comment:
Please try to handle any problems in the model so the error will originate from there and ends up in your error handling middleware (in app.js) by going through the catch block in your controller.